### PR TITLE
Improve .gitignore - modern comment-based template

### PR DESCRIPTION
**Pull request by dzikoysk**
Consistent with the official rules
* https://github.com/github/gitignore/blob/master/Java.gitignore
* https://www.gitignore.io/

### CODE REVIEW

**Comment by dzikoysk** [comment]
`/lib` is not excluded. By the way, you should install these libraries in repo, keeping dependencies with a project is not a good idea

**Comment by dzikoysk** [comment]
Actually, you should install these libraries in repo, keeping dependencies with a project is not a good idea

**Comment by bensku** [comment]
I do not compile them myself, so I'm not sure how that would work. I use terrible hack for providing Skript's Maven repo at the moment...

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,114 @@
...
+
+# JIRA plugin
+atlassian-ide-plugin.xml
+
+# Crashlytics plugin (for Android Studio and IntelliJ)
+com_crashlytics_export_strings.xml
+crashlytics.properties
+crashlytics-build.properties
+
+### Eclipse ###
```
Comment (bensku) [inline .gitignore]: Skript relies on most Eclipse settings being on the repo. They are used to make sure nullness analysis works correctly; that includes command-line compilation.

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,114 @@
...
+.externalToolBuilders/
+
+# Locally stored "Eclipse launch configurations"
+*.launch
+
+# CDT-specific
+.cproject
+
+# JDT-specific (Eclipse Java Development Tools)
+.classpath
```
Comment (bensku) [inline .gitignore]: Might be necessary to have in repo. Not 100% sure.

File: .gitignore [hunk 3]
```diff
@@ -1,7 +1,114 @@
...
+*.tmp
+*.bak
+*.swp
+*~.nib
+local.properties
+.settings/
+.loadpath
+
+# Eclipse Core
+.project
```
Comment (bensku) [inline .gitignore]: Might be necessary to have in repo. Not 100% sure.

**Comment by bensku** [comment]
I don't know why I did not notice the few other potential problems. Sorry for delay, but they must be fixed before merging; stuff might break quite badly if that is not done.

File: .gitignore (hunk 2 above)
Comment (Syst3ms) [reply .gitignore]: IntelliJ can technically use those to create the project, but it can also use build.gradle

Comment (bensku) [reply .gitignore]: I meant that Eclipse may need this, but not 100% sure since this is Gradle project.

Comment (Syst3ms) [reply .gitignore]: Better safe than sorry, I'd say to keep it

**Comment by dzikoysk** [comment]
The longer I look at it the less I want to continue touching it. I recommend to close the pr because in fact, it is the eclipse-based project, not a specific build tool.

**Pull request closed without merging**